### ISSUE TITLE
Running error: can't run linter goanalysis_metalinter

### ISSUE THREAD

**Issue by QZAiXH** [golangci/golangci-lint#5184]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

```
golangci-lint run --fix
WARN [runner] Can't run linter goanalysis_metalinter: inspect: failed to load package app: could not load export data: no export data for "fyne.io/fyne/v2/app"
ERRO Running error: can't run linter goanalysis_metalinter
inspect: failed to load package app: could not load export data: no export data for "fyne.io/fyne/v2/app"
```
**I skipped the directory where that file is located and I still get the same error.**

### Version of golangci-lint

```
golangci-lint --version
golangci-lint has version v1.62.2 built with go1.23.2 from (unknown, modified: ?, mod sum: "h1:b8K5K9PN+rZN1+mKLtsZHz2XXS9aYKzQ9i25x3Qnxxw=") on (unknown)
```

### Configuration

<details>

```yml
run:
  timeout: 5m

linters:
  disable-all: true
  enable:
    - gofumpt
    - govet
    - misspell
    - revive
    - lll

linters-settings:
  gofumpt:
    extra-rules: true

  lll:
    line-length: 160
    tab-width: 1

  govet:
    # Matches what we use in corp as of 2023-12-07
    enable:
      - asmdecl
      - assign
      - atomic
      - bools
      - buildtag
      - cgocall
      - copylocks
      - deepequalerrors
      - errorsas
      - framepointer
      - httpresponse
      - ifaceassert
      - loopclosure
      - lostcancel
      - nilfunc
      - nilness
      - printf
      - reflectvaluecompare
      - shift
      - sigchanyzer
      - sortslice
      - stdmethods
      - stringintconv
      - structtag
      - testinggoroutine
      - tests
      - unmarshal
      - unreachable
      - unsafeptr
      - unusedresult

  revive:
    enable-all-rules: false
    ignore-generated-header: true
    rules:
      - name: atomic
      - name: context-keys-type
      - name: defer
        arguments: [ [
          # Calling 'recover' at the time a defer is registered (i.e. "defer recover()") has no effect.
          "immediate-recover",
          # Calling 'recover' outside of a deferred function has no effect
          "recover",
          # Returning values from a deferred function has no effect
          "return",
        ] ]
      - name: duplicated-imports
      - name: errorf
      - name: string-of-int
      - name: time-equal
      - name: unconditional-recursion
      - name: useless-break
      - name: waitgroup-by-value

issues:
  exclude-dirs:
    - daemon-ui
```

</details>

### Go environment

<details>

```
go version && go env
go version go1.23.2 linux/amd64
GO111MODULE='on'
GOARCH='amd64'
GOBIN=''
GOCACHE='/root/.cache/go-build'
GOENV='/root/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/root/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/root/go'
GOPRIVATE=''
GOPROXY='https://goproxy.cn,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.23.2'
GODEBUG=''
GOTELEMETRY='local'
GOTELEMETRYDIR='/root/.config/go/telemetry'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='0'
GOMOD='/mnt/i/code/mesh/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -fno-caret-diagnostics -Qunused-arguments -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build3840113770=/tmp/go-build -gno-record-gcc-switches'
```

</details>

### Verbose output of running

<details>

```
golangci-lint run -v
INFO golangci-lint has version v1.62.2 built with go1.23.2 from (unknown, modified: ?, mod sum: "h1:b8K5K9PN+rZN1+mKLtsZHz2XXS9aYKzQ9i25x3Qnxxw=") on (unknown)
INFO [config_reader] Config search paths: [./ /mnt/i/code/mesh /mnt/i/code /mnt/i /mnt / /root]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 5 linters: [gofumpt govet lll misspell revive]
INFO [loader] Go packages loading at mode 8767 (compiled_files|exports_file|imports|name|deps|files|types_sizes) took 18.030615921s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 438.588443ms
INFO [linters_context/goanalysis] analyzers took 33.125506682s with top 10 stages: the_only_name: 22.572948634s, gofumpt: 2.749654897s, misspell: 2.255548414s, lll: 2.153301417s, inspect: 1.292751971s, printf: 704.465039ms, ctrlflow: 666.508529ms, buildssa: 397.758806ms, lostcancel: 42.964364ms, directive: 32.50342ms
WARN [runner] Can't run linter goanalysis_metalinter: inspect: failed to load package app: could not load export data: no export data for "fyne.io/fyne/v2/app"
INFO [runner] processing took 2.78µs with stages: max_same_issues: 510ns, path_prettifier: 460ns, skip_dirs: 290ns, cgo: 230ns, skip_files: 140ns, filename_unadjuster: 140ns, invalid_issue: 130ns, nolint: 130ns, max_from_linter: 120ns, identifier_marker: 110ns, fixer: 50ns, max_per_file_from_linter: 50ns, diff: 50ns, exclude: 50ns, sort_results: 40ns, exclude-rules: 40ns, autogenerated_exclude: 40ns, uniq_by_line: 40ns, severity-rules: 40ns, path_shortener: 40ns, source_code: 40ns, path_prefixer: 40ns
INFO [runner] linters took 6.141458296s with stages: goanalysis_metalinter: 6.141355877s
ERRO Running error: can't run linter goanalysis_metalinter
inspect: failed to load package app: could not load export data: no export data for "fyne.io/fyne/v2/app"
INFO Memory: 248 samples, avg is 175.5MB, max is 753.3MB
INFO Execution took 24.662479151s
```

</details>

### A minimal reproducible example or link to a public repository

```
package main

import (
	"fyne.io/fyne/v2/app"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello World!")
	w.ShowAndRun()
}

```

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

you should run `go mod tidy`.

``console
$ golangci-lint run                                           
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "fyne.io/fyne/v2/app" 
ERRO Running error: can't run linter goanalysis_metalinter
buildir: failed to load package : could not load export data: no export data for "fyne.io/fyne/v2/app" 
$ go mod tidy                          
go: finding module for package fyne.io/fyne/v2/app
go: downloading fyne.io/fyne v1.4.3
go: downloading fyne.io/fyne/v2 v2.5.2
go: found fyne.io/fyne/v2/app in fyne.io/fyne/v2 v2.5.2
go: downloading github.com/rymdport/portal v0.2.6
go: downloading github.com/fredbi/uri v1.1.0
go: downloading fyne.io/systray v1.11.0
go: downloading github.com/fyne-io/gl-js v0.0.0-20220119005834-d2da28d9ccfe
go: downloading github.com/fyne-io/glfw-js v0.0.0-20240101223322-6e1efdc71b7a
go: downloading github.com/fyne-io/image v0.0.0-20220602074514-4956b0afb3d2
go: downloading github.com/go-gl/glfw/v3.3/glfw v0.0.0-20240506104042-037f3cc74f2a
go: downloading github.com/go-text/typesetting v0.2.0
go: downloading github.com/go-text/render v0.2.0
go: downloading github.com/srwiley/rasterx v0.0.0-20220730225603-2ab79fcdd4ef
go: downloading github.com/srwiley/oksvg v0.0.0-20221011165216-be6e8873101c
go: downloading github.com/go-gl/gl v0.0.0-20211210172815-726fda9656d6
go: downloading golang.org/x/mobile v0.0.0-20231127183840-76ac6878050a
go: downloading github.com/jeandeaual/go-locale v0.0.0-20240223122105-ce5225dcaa49
go: downloading github.com/jsummers/gobmp v0.0.0-20151104160322-e2ba15ffa76e
go: downloading github.com/pkg/profile v1.7.0
go: downloading github.com/felixge/fgprof v0.9.3
go: downloading github.com/go-text/typesetting-utils v0.0.0-20240317173224-1986cbe96c66
go: downloading github.com/google/pprof v0.0.0-20211214055906-6f57359322fd
$ golangci-lint run --show-stats
0 issues.

```

**Comment by QZAiXH** [comment]
This error triggers in Linux environments, but runs fine in Windows. It also occurs when I run it on GitHub Actions. The configuration is as follows:

```yaml
name: lint

on:
  pull_request:

permissions:
  contents: read
  pull-requests: read

concurrency:
  group: ${{ github.workflow }}-$${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

jobs:
  lint:
    name: Run lint
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4

    - uses: actions/setup-go@v5
      with:
        go-version-file: go.mod
        cache: true

    - name: golangci-lint
      uses: golangci/golangci-lint-action@v6
      with:
        version: latest
        args: --timeout=10m
        only-new-issues: true
```

The error message is as follows:

```
prepare environment
  Checking for go.mod: go.mod
  Cache not found for input keys: golangci-lint.cache-Linux-2865-b08b7b01925ca925eca055fbccf3ca923c38d824, golangci-lint.cache-Linux-2865-
  Finding needed golangci-lint version...
  Requested golangci-lint 'latest', using 'v1.62.2', calculation took 21ms
  Installation mode: binary
  Installing golangci-lint binary v1.62.2...
  Downloading binary https://github.com/golangci/golangci-lint/releases/download/v1.62.2/golangci-lint-1.62.2-linux-amd64.tar.gz ...
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: inspect: failed to load package app: could not load export data: no export data for \"fyne.io/fyne/v2/app\""
  level=error msg="Running error: can't run linter goanalysis_metalinter\ninspect: failed to load package app: could not load export data: no export data for \"fyne.io/fyne/v2/app\""
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 108682ms
```

**Comment by ldez** [comment]
You should try to remove `only-new-issues: true`

**Comment by QZAiXH** [comment]
Thanks for your reply. Before you replied I tried to change `runs-on` to `windows-latest` and this time it was executing fine!

**Comment by QZAiXH** [comment]
> You should try to remove `only-new-issues: true`

I changed `runs-on` back to `ubuntu-latest` and deleted `only-new-issues: true` and ran it again, and got the same error, which is now guaranteed to be reproduced in `ubuntu`.

**Comment by ldez** [comment]
So the problem is related to your project: you have to check your `go.mod`, run `go mod tidy`, check the `go.sum`.
